### PR TITLE
Update fastify to v3

### PR DESCRIPTION
**Pull request by frikille**
* Updates fastify to v3
* Adds dependabot config
* Moves CI from Travis to Github Actions

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .dependabot/config.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+version: 1
+update_conigs:
+  package_manager: "javascript"
+  directory: "/"
+  update_schedule: "daily"
```
Comment (mcollina) [inline .dependabot/config.yml:5]: this is not correct, check the other ones I had to fix manually :)

**Comment by mcollina** [comment]
@Eomm can you take a look?

File: test/transform.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ test('transform should fail with a value other than Function', t => {
 
   fastify.register(fastifySwagger, invalid)
 
-  fastify.setSchemaCompiler(schema => Joi.validate(schema))
+  fastify.setValidatorCompiler(schema => Joi.validate(schema))
```
Comment (Eomm) [inline test/transform.js:37]:
```suggestion
  fastify.setValidatorCompiler(({ schema }) => Joi.validate(schema))
```

File: test/transform.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ test('transform should work with a Function', t => {
 
   fastify.register(fastifySwagger, valid)
 
-  fastify.setSchemaCompiler(schema => Joi.validate(schema))
+  fastify.setValidatorCompiler(schema => Joi.validate(schema))
```
Comment (Eomm) [inline test/transform.js:51]:
```suggestion
  fastify.setValidatorCompiler(({ schema }) => Joi.validate(schema))
```

**Comment by Eomm** [comment]
@mcollina I need this merge to complete #239

**Comment by mcollina** [comment]
landed

**Pull request merged**